compiler/src/dotty/tools/dotc/typer/ProtoTypes.scala

```diff
@@ -418,17 +428,17 @@ object ProtoTypes {
       isInstantiatable(tp) || tp.member(nme.apply).hasAltWith(d => isInstantiatable(d.info))
     }
 
-    def derivedPolyProto(targs: List[Type], resultType: Type): PolyProto =
+    def derivedPolyProto(targs: List[Tree], resultType: Type): PolyProto =
       if ((targs eq this.targs) && (resType eq this.resType)) this
       else PolyProto(targs, resType)
 
     override def notApplied: Type = WildcardType
 
     def map(tm: TypeMap)(implicit ctx: Context): PolyProto =
-      derivedPolyProto(targs mapConserve tm, tm(resultType))
+      derivedPolyProto(targs, tm(resultType))
```

Comment:
Hmm what guarantee do we have that not mapping over the args is OK here ?

Reply:
The intuition is to be analogous to FunProto where argument types are also not transformed. The arguments of a PolyProto are conceptually program trees, which are not transformed, we only store them as types because we don't need the tree information.